rpc/legacyrpc/methods.go

```diff
@@ -967,23 +946,7 @@ func createNewAccount(icmd interface{}, w *wallet.Wallet) (interface{}, error) {
 		return nil, &ErrReservedAccountName
 	}
 
-	// Check that we are within the maximum allowed non-empty accounts limit.
-	account, err := w.Manager.LastAccount()
-	if err != nil {
-		return nil, err
-	}
-	if account > maxEmptyAccounts {
-		used, err := w.AccountUsed(account)
-		if err != nil {
-			return nil, err
-		}
-		if !used {
-			return nil, errors.New("cannot create account: " +
-				"previous account has no transaction history")
-		}
-	}
-
-	_, err = w.NextAccount(cmd.Account)
+	_, err := w.NextAccount(cmd.Account)
```

Comment:
Well, that's not totally true.  If it's non-nil but not waddrmgr.ErrLocked, then it will still be returned with whatever the wallet fallback error code is.